### DISCUSSION TITLE
pytest on Github Actions fails when shared among Linux and Windows

### DISCUSSION THREAD

**Discussion by brunolnetto** [pytest-dev/pytest#10462] [Q&A] [answered]
Hi fellows,

I implement a GitHub Action to cover all OS we know (Linux, macOS, Windows). GitHub actions offer the options matrix/strategy to select locally defined variables. However, I fail to use the if clause to choose between different commands.  Its definition is on the URL below: 

https://github.com/quivero/eule/blob/main/.github/workflows/test.yml

The `codecov` job comprehends 3 steps:

1. Setup Python;
2. Generate `codecov` report;
3. Upload coverage to Codecov;

The Windows command run fails to execute the command `pytest --cov=./ --cov-report=xml`. It must be something like `python pytest --cov=./ --cov-report=xml` for `${{ matrix.os }}` equals to `windows-latest`. 

All necessary error logs are available on the last workflow run of URL below:

https://github.com/quivero/eule/actions/runs/3379687126

**Comment by nicoddemus** [comment, marked as answer]
Hey,

I would try to always execute with `python` in both platforms:

```
      run: |
        python -m pip install -r requirements_dev.txt  
        python -m pytest --cov=./ --cov-report=xml  
```

you cannot hard code the path to ` c:\hostedtoolcache\windows\python\3.7.9\x64\python.exe` given that you are setting up multiple Python versions. The `setup-python` action takes care of putting the right executable on `PATH`, so just calling `python` should work.

**Comment by brunolnetto** [comment]
Your name is funny and made me laugh. Btw, the solution works. Thanks.
